src/source/Common/AwsV4Signer.c

```diff
@@ -498,8 +510,14 @@ STATUS generateCanonicalRequestString(PRequestInfo pRequestInfo, PCHAR pRequestS
     len = SHA256_DIGEST_LENGTH * 2;
     CHK(curLen + len <= requestLen, STATUS_BUFFER_TOO_SMALL);
     if (pRequestInfo->body == NULL) {
-        // Streaming treats this portion as if the body were empty
-        CHK_STATUS(hexEncodedSha256((PBYTE) EMPTY_STRING, 0, pCurPtr));
+        if (!STRNCMP(pRequestInfo->service, KINESIS_VIDEO_SERVICE_NAME, MAX_SERVICE_NAME_LEN)) {
+            // Streaming treats this portion as if the body were empty
+            CHK_STATUS(hexEncodedSha256((PBYTE) EMPTY_STRING, 0, pCurPtr));
+        } else {
+            PCHAR term = "UNSIGNED-PAYLOAD";

```

Comment:
This should be defined in the internal header. The length in that case should be calculated as ARRAY_SIZE(PREDEFINED_UNSIGNED_PAYLOAD)